tests/integration/test_bayesian_optimization.py

```diff
@@ -38,7 +38,6 @@
 from tests.util.misc import random_seed
 
 
-@gpflow.config.as_context(gpflow.config.Config(float=tf.float64))
```

Comment:
this is no longer working. Don't know why it changed - there's no obvious related change in gpflow. The test also passes without it (it was added before to avoid cholesky errors)

Reply:
I guess this just set the default float to 64bit for the purposes of the test. Are the floats 64bit anyway?

Reply:
(it probably is a good idea to make sure the tests are using 64bit floats)

Reply:
ah, gpflow's default float is float64, I thought it was float32

as an aside, why do you reckon we should use float64 for tests? I'd read somewhere (not authoritative) saying float32 was the standard. `tf.constant(1.0).dtype` is `tf.float32` but `np.array(1.0).dtype` is `np.float64`. Rather odd

Reply:
for neural nets you want to minimise memory usage, and the difference between 32 and 64 bit isn't very significant. For GPs often suffer from a lot of numerical instability issues which are reduced by the extra precision - and normally the memory footprint isn't something we have to worry about.

Reply:
I'd guess the numpy vs tensor thing hits the same issue - numpy more people face numerical instability issues, tensorflow more people memory issues